src/guides/v2.3/migration/extend-the-tool.md

```diff
@@ -121,7 +152,245 @@ Steps can include four types of classes:
+
+For Integrity checking we define additional map file in `config.xml`. It will be used to verify that tables structure is like we expect.
+
+```xml
+<config xmlns:xs="http://www.w3.org/2001/XMLSchema-instance"
+        xs:noNamespaceSchemaLocation="urn:magento:module:Magento_DataMigrationTool:etc/config.xsd">
+    ...
+    <options>
+        ...
+        <greatblog_map_file>app/code/Vendor/Migration/etc/opensource-to-opensource/map-greatblog.xml</greatblog_map_file>
+        ...
+    </options>
+</config>
+```
+
+Map file `map-greatblog.xml`:
+
+```xml
+<map xmlns:xs="http://www.w3.org/2001/XMLSchema-instance"
+     xs:noNamespaceSchemaLocation="urn:magento:module:Magento_DataMigrationTool:etc/map.xsd">
+    <source>
+        <field_rules>
+            <ignore>
+                <field>greatblog_post.tags</field>
+            </ignore>
+        </field_rules>
+    </source>
+    <destination>
+        <document_rules>
+            <ignore>
+                <document>greatblog_post_tags</document>
+            </ignore>
+        </document_rules>
+    </destination>
+</map>
+```
+
+Integrity checking class extends `Migration\App\Step\AbstractIntegrity` and contains `perform` method where we verify tables structure:
+
+```php
+class Integrity extends \Migration\App\Step\AbstractIntegrity
+{
+    ...
+    /**
+     * Integrity constructor.
+     * @param ProgressBar\LogLevelProcessor $progress
+     * @param Logger $logger
+     * @param Config $config
+     * @param ResourceModel\Source $source
+     * @param ResourceModel\Destination $destination
+     * @param MapFactory $mapFactory
+     * @param string $mapConfigOption
+     */
+    public function __construct(
+        ProgressBar\LogLevelProcessor $progress,
+        Logger $logger,
+        Config $config,
+        ResourceModel\Source $source,
+        ResourceModel\Destination $destination,
+        MapFactory $mapFactory,
+        $mapConfigOption = 'greatblog_map_file'
+    ) {
+        parent::__construct($progress, $logger, $config, $source, $destination, $mapFactory, $mapConfigOption);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function perform()
+    {
+        $this->progress->start($this->getIterationsCount());
+        $this->check(['greatblog_post'], MapInterface::TYPE_SOURCE);
+        $this->check(['greatblog_post', 'greatblog_post_tags'], MapInterface::TYPE_DEST);
+        $this->progress->finish();
+        return $this->checkForErrors();
+    }
+    ...
+}
+```
+
+Next you will need to create a class for processing and saving data to Magento 2 database `Vendor\Migration\Step\GreatBlog\Data`:
+
+```php
+class Data implements StageInterface
```

Comment:
We may provide the fully qualified class name for `StageInterface` because the namespace might not be obvious from the alias.

Reply:
The same suggestion applies to other similar examples

Reply:
Changes have been made. Thank you